File: C++/1726. Tuple with Same Product.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& n) {
        if(n.size()<4)
        return 0;
        map<int,int>m;
        int c=0;
        for(int i=0;i<n.size();i++){
            for(int j=i+1;j<n.size();j++){
                m[n[i]*n[j]]++;
            }
        }
        for(auto i=m.begin();i!=m.end();i++){
            c+=8*((i->second-1)*i->second/2);
        }
        return c;

    }
};

File: JAVA/1726. Tuple with Same Product.java
class Solution {
    public int tupleSameProduct(int[] n) {
        if (n.length < 4) return 0;
        Map<Integer, Integer> m = new HashMap<>();
        int c = 0;
        for (int i = 0; i < n.length; i++) {
            for (int j = i + 1; j < n.length; j++) {
                m.put(n[i] * n[j], m.getOrDefault(n[i] * n[j], 0) + 1);
            }
        }
        for (int v : m.values()) {
            c += 8 * (v * (v - 1) / 2);
        }
        return c;
    }
}